Code to select the aod motion computation method based on file versio…

saumilpatel opened the pull request:
…n. Supports file version in which motion data are 3D

Commit:
Volume ignore improvement

peabody124 committed:
AOD: switch some objects to dj.AutoPopulate

peabody124 committed:
AOD: switch some objects to dj.AutoPopulate

peabody124 committed:
AOD: check in some changes laying around

peabody124 committed:
AOD: check in some changes laying around

peabody124 committed:
Merge branch 'master' of github.com:peabody124/sessions

Commit:
Merge branch 'master' of github.com:peabody124/sessions

Commit:
Whatever James had not commited...

eywalker committed:
Fix bug where running new spike detection method deletes previous detected spikes data

eywalker committed:
SyncNetworkFewTrials: check in missing file

Commit:
Code to select the aod motion computation method based on file version. Supports file version in which motion data are 3D

dimitri-yatsenko commented on startup.m:
```diff
@@ -0,0 +1,6 @@
+setPath ;
+setenv('DJ_HOST', 'at-database.ad.bcm.edu');
+setenv('DJ_USER','spatel');
+setenv('DJ_PASS','spatel');
```
argh!  checked in password.  Please change your password and exclude your `startup.m` from the path or set the environment variables in a separate script such as your `~/.profile` (on Mac OS)

saumilpatel replied:
argh!, removed the startup from the location where it was stored, moved the credentials to matlab's default startup on the machine, also changed password

peabody124 replied:
it is probably worth doing a squash of these two commits together and then doing a force push to your repository to overwrite them, so when the code is merged that never shows up in the history

dimitri-yatsenko commented on schemas/+aod/trackMotion3D.m:
```diff
@@ -0,0 +1,91 @@
+analyzenthframe = 1 ; % skip frames if > 1
+numframestoaverageforref = 10 ; % frames to average to determine the reference frame against which motion is computed
+usemulticore = true ;
+postrefstartframetime = 0 ; % N secs to skip before analyzing motion signals, in many cases, right after starting the point scan, there is a slight drift
```
for better readability, use camelCase or underscore_separated compound names.

dimitri-yatsenko commented on schemas/+aod/trackMotion3D.m:
```diff
@@ -0,0 +1,91 @@
+    count = 0 ;
+    for ii=numframestoaverageforref+analyzenthframe:analyzenthframe:size(dat2,1)
+        count = count + 1 ;
+        for jj=ii-analyzenthframe+1:ii
```
you can replace this loop with array operations:
```matlab
adat(count,:) = adat(count, :) + mean(dat2(ii-analyzenthframe+1:ii,:))
```

saumilpatel replied:
replaced

dimitri-yatsenko commented on schemas/+aod/trackMotion3D.m:
```diff
@@ -0,0 +1,91 @@
+
+% take into account frames to skip, i.e. sampling rate of motion,
+% downsample with averaging
+adat = zeros(size(dat2)) ;
```
replace these loops with matlab's implementation of downsampling.

saumilpatel replied:
did not change, cant find downsampling code that will average samples, dont want to deal with filters

dimitri-yatsenko commented on schemas/+aod/trackMotion3D.m:
```diff
@@ -0,0 +1,91 @@
+needreference=false ; % no need for a reference, start computing positions relative to the reference of spheres in subsequent motion frames
+if (usemulticore)
+    parfor ii=1:count
+        [shifts(ii+numframestoaverageforref,:)] = computeshifts(adat(ii,:)',coordinates,gridsize,calibparams,refparams,needreference) ;  
```
don't need brackets `[  ... ]` around `shifts(...)`

saumilpatel replied:
removed

dimitri-yatsenko commented on schemas/+aod/trackMotion3D.m:
```diff
@@ -0,0 +1,91 @@
+adat = zeros(size(dat2)) ;
+if (analyzenthframe>1)
+    count = 0 ;
+    for ii=numframestoaverageforref+analyzenthframe:analyzenthframe:size(dat2,1)
```
This loop may ignore the tail of the `dat2` when `analyzenthframe>1`. Is that intended?

saumilpatel replied:
dont follow, but my best guess is what you mean is that samples at the end may not be processed, ?

dimitri-yatsenko commented on schemas/+aod/trackMotion3D.m:
```diff
@@ -0,0 +1,91 @@
+numsamples = br1.sz ;
+duration = numsamples(:,1)/br1.Fs ; % sec
+
+%numframes = size(dat,1) ;
```
remove commented code.

saumilpatel replied:
removed

dimitri-yatsenko commented on schemas/+aod/trackMotion3D.m:
```diff
@@ -0,0 +1,91 @@
+% requires z:\libraries\matlab in path
+run(getLocalPath('/lab/libraries/hdf5matlab/setPath.m')) ;
+
+analyzenthframe = 1 ; % skip frames if > 1
```
Rename `analyzenthframe` to something like `downsampleFactor`?

saumilpatel replied:
renamed

Commit:
1. changed variables names
2. replaced downsampling inner loop with vector operation
3. removed commented code
4. matlab style changes
Nothing done here affects functionality

peabody124 committed:
Work remotely

peabody124 committed:
Merge branch 'master' of https://github.com/saumilpatel/sessions into rebase

peabody124 commented on schemas/+aod/ScanMotion.m:
```diff
@@ -27,16 +27,24 @@ function makeTuples(self, key)
-            coordinates = br.motionCoordinates;
-            
+            fn = char(fetchn(acq.AodScan & key, 'aod_scan_filename')) ; % get the filename from db
+            oldversion = aod.checkFileversion(fn) ; % check if motion data is in old format, i.e. 2 planes or in new format, i.e. a volume
```
this is a typo, or more likely the checkFileVersion should have been checked into the aod directory instead of the acq directory.

saumilpatel replied:
oops... yes a typo, should be acq.checkFileVersion

peabody124 replied:
it's probably best to keep with the aod schema since it doesn't apply to anything else in acq. Also, if we can refactor it into the AodMotionReader class so that this isn't exported in general that would be better (see comment below)

saumilpatel replied:
agree... will move it

peabody124 commented on schemas/+aod/trackMotion3D.m:
```diff
@@ -0,0 +1,86 @@
+
+%% test function computeshifts
+% requires z:\libraries\matlab in path
+run(getLocalPath('/lab/libraries/hdf5matlab/setPath.m')) ;
```
this shouldn't be here. the user should have set up the path correctly before hand. this would likely have unintended side effects.

saumilpatel replied:
agree, will remove

peabody124 commented on schemas/+acq/checkFileversion.m:
```diff
@@ -0,0 +1,18 @@
+function oldversion=checkFileVersion(fn)
+
+oldversion = true ;
+try
```
this is failing for me when analyzing 

    M/Mouse/2016-03-15_13-56-05/AODAcq/2016-03-15_14-38-45_%d.h5

specifically the h5readatt fails so we fall through to the catch statement. i believe this experiment was recorded with the new format, though.

saumilpatel replied:
Huh... works fine for the file you mention on at-ssp, Matlab R2016b. Is h5readatt working for other files ?

peabody124 replied:
Can you check it from a mac? i'm using this from my local machine (R2016a but that command was introduced in 2011a)

saumilpatel replied:
Yeah, h5readatt does not work on my Mac 2016a... let me check why

saumilpatel replied:
it works if the file name is /Volumes/M....
Is this what you meant in the comment getFile... To translate the file name for the platform used ?

peabody124 commented on schemas/+aod/trackMotion3D.m:
```diff
@@ -0,0 +1,86 @@
+useMultiCore = true ;
+postRefStartFrameTime = 0 ; % N secs to skip before analyzing motion signals, in many cases, right after starting the point scan, there is a slight drift
+
+br = aodReader(fn, 'Motion') ;
```
here I think this should use getFile(fn, 'Motion') to convert  the path from /raw/blah to the real location

saumilpatel replied:
this I am not following...

peabody124 replied:
I ended up replacing this with

    fn2 = findFile(RawPathMap, fn)
    br = aodReader(fn2, 'Motion');
    br1 = AodMotionCorrectionReader(fn2) ; % if motion was corrected by using aod offsets, this is where they would be read

saumilpatel replied:
sounds good

peabody124 commented on schemas/+aod/trackMotion3D.m:
```diff
@@ -0,0 +1,86 @@
+calibparams = [0 1; 0 1; 0 1] ; % dont apply any calibration
+shifts = zeros(size(dat2,1),3); % store motion vectors here
+dat2(1,:) = mean(dat2(1:numFramesToAverageForRef,:)) ; % average frames for computing reference
+[shifts(1,:), refparams] = computeshifts(dat2(1,:)',coordinates,gridSize,calibparams,refparams,needReference) ; % this position of the fitted sphere is treated as the reference
```
this function is missing. above it references it being in lab\libraries\matlab but i don't see it there. also, please let's not spread these dependencies across multiple repositories.

saumilpatel replied:
This is in z:\libraries\TwoPhoton\AOD_Control\MotionCorrection. Note that we are using the same motion detection code that is used from the Labview program during real-time acquisition.

peabody124 replied:
I don't really like duplicating code in two places, but at the same time in this case I think it would be better than spanning dependencies between two pretty separate projects. It also avoids unintended side effects if one of the formats changes the other would break.

saumilpatel replied:
OK, would make the sessions folder self contained but we will have to remember to keep the two folders synced.

peabody124 commented on schemas/+aod/trackMotion3D.m:
```diff
@@ -0,0 +1,86 @@
+postRefStartFrameTime = 0 ; % N secs to skip before analyzing motion signals, in many cases, right after starting the point scan, there is a slight drift
+
+br = aodReader(fn, 'Motion') ;
+br1 = AodMotionCorrectionReader(fn) ; % if motion was corrected by using aod offsets, this is where they would be read
```
do we need a separate class here just to access the sampling rate? can we not expand the original AodMotionReader to handle both formats correctly? I only see br1 used below to access the number of samples and sampling rate, which AodMotionReader should just be taught to do correctly for this new format.

saumilpatel replied:
Probably not but we are just reusing code that we have.

peabody124 commented on schemas/+aod/trackMotion3D.m:
```diff
@@ -0,0 +1,86 @@
+
+function [x,y,z,t] = trackMotion3D(fn, analysischan) 
```
could we refactor this a bit so move all the code that deals with the raw file access into ScanMotion (like it is for the original code) and then make this method just take the data set and process it? that would keep the two consistent.

saumilpatel replied:
We have to run motion detection during realtime acq as well as during offline analysis. Because realtime implementation came first, the code is residing in folder structure that we have for realtime stuff. We can rearrange if it is absolutely necessary, for sure we dont want to duplicate the same code in different folders

Commit:
Changed location of checkFilename from +acq to +aod
Removed path setup from within trackMotion3D

peabody124 committed:
Check in code for Saumil's motion tracking

peabody124 committed:
detectLcdPhotodiodeFlips: get rid of external dependecy

eywalker committed:
Small fixes

eywalker committed:
AOD: improve robustness of syuncing

eywalker committed:
Merge remote-tracking branch 'origin/james'

Conflicts:
	james_setPath.m
	processing/cleanup.m
	processing/spikesUtah.m

eywalker committed:
Make RawPathMap updatable

eywalker committed:
Adjust diode offset accepted range

eywalker committed:
Add special handling for NatImExperiment end time

eywalker committed:
Merge pull request #17 from peabody124/james

Incorporate dangling changes and updates to AOD schema

saumilpatel committed:
Merge pull request #14 from atlab/bugfix

Fix issue #13

eywalker committed:
Merge branch 'master' of https://github.com/atlab/sessions

Conflicts:
	schemas/+ephys/SpikesAlignedTrial.m

saumilpatel committed:
code to record notification sent to slack in database
updated session cleanup code
store results of saccadeflashexperiment to database

saumilpatel committed:
Merge branch 'master' of https://github.com/saumilpatel/sessions

saumilpatel committed:
added /Volumes/M to search path

saumilpatel committed:
NotificationSent table moved from acq to stimulation
